query-engine/core/src/executor/request_context.rs

```diff
@@ -18,14 +18,13 @@ tokio::task_local! {
 ///
 /// If we had a query context we carry for the entire lifetime of the query, it would belong there.
 pub(crate) fn get_request_now() -> PrismaValue {
-    // FIXME: we want to bypass task locals if this code is executed outside of a tokio context. As
-    // of this writing, it happens only in the query validation test suite.
-    //
-    // Eventually, this will go away when we have a plain query context reference we pass around.
-    if tokio::runtime::Handle::try_current().is_err() {
-        return PrismaValue::DateTime(chrono::Utc::now().into());
-    }
-    REQUEST_CONTEXT.with(|rc| rc.request_now.clone())
+    REQUEST_CONTEXT.try_with(|rc| rc.request_now.clone()).unwrap_or_else(|_|
+            // FIXME: we want to bypass task locals if this code is executed outside of a tokio context. As
+            // of this writing, it happens only in the query validation test suite.
+            //
+            // Eventually, this will go away when we have a plain query context reference we pass around.
+            // Skipping the branch for WASM since there we never have a running tokio runtime
```

Comment:
I think by now the comments are more confusing than useful, and I would be more comfortable removing them.

Reply:
I still think there are useful bits in there, reworded to bring it up to date and make less confusing.